zanata-client-commands/src/main/java/org/zanata/client/commands/pull/PullOptionsImpl.java

```diff
@@ -166,6 +167,16 @@ public boolean isContinueAfterError() {
     }
 
     @Override
+    public int getMinDocPercent() {
+        return this.minDocPercent;
+    }
+
+    @Option(name = "--min-doc-percent", aliases = "-m", usage = "When specified, will only pull down translation files that have translated or better statistics above the specified percentage")
```

Comment:
"Only pull documents which are at least PERCENT % translated" along with a
metaVar for PERCENT. Users won't care about the states, they just care
whether it has been translated.  When we implement the feature you mention,
we can update the description to cover Approved, probably along with the
option to override it, etc etc.

I guess if we want the same option also to work for "at least PERCENT %
approved" later on, it might be better to keep the option name generic,
like "--min-doc-percent".

On 19 May 2015 at 13:50, Patrick Huang <notifications@github.com> wrote:

> In
> zanata-client-commands/src/main/java/org/zanata/client/commands/pull/PullOptionsImpl.java
> <https://github.com/zanata/zanata-client/pull/63#discussion_r30567527>:
>
> > @@ -166,6 +167,16 @@ public boolean isContinueAfterError() {
> >      }
> >
> >      @Override
> > +    public int getMinDocPercent() {
> > +        return this.minDocPercent;
> > +    }
> > +
> > +    @Option(name = "--min-doc-percent", aliases = "-m", usage = "When specified, will only pull down translation files that have translated or better statistics above the specified percentage")
>
> Do you have a better sentence in mind? I couldn't work out how to convey
> the meaning of translated and approved state. Also when we implement the
> feature where project's requiring review will only allow users to download
> approved state, this sentence will still make sense.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-client/pull/63/files#r30567527>.
>